sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -42,10 +42,12 @@ trait LimitExec extends UnaryExecNode {
  * This operator will be used when a logical `Limit` operation is the final operator in an
  * logical plan, which happens when the user is collecting results back to the driver.
  */
-case class CollectLimitExec(limit: Int, child: SparkPlan) extends LimitExec {
+case class CollectLimitExec(limit: Int, offset: Int, child: SparkPlan) extends LimitExec {
   override def output: Seq[Attribute] = child.output
   override def outputPartitioning: Partitioning = SinglePartition
-  override def executeCollect(): Array[InternalRow] = child.executeTake(limit)
+  override def executeCollect(): Array[InternalRow] = {
+    child.executeTake(limit + offset).drop(offset)
```

Comment:
can you add a comment mentioning the reasoning behind why we add the `limit` and `offset` together into a single sum here, and then drop the `offset`, with a brief example?

Reply:
OK

Reply:
Can we avoid the `.drop` if `offset == 0`? `.drop` creates a new collection instance.

Reply:
OK